File: Data Structures & Algorithms/maximum-subarray/submission-1.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = nums[0];
        int currSum = 0;

        for (auto num : nums)
        {
            currSum = max(currSum, 0);
            currSum += num;
            maxSum = max(maxSum, currSum);
        }

        return maxSum;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = {}


        for num in nums:
            counts[num] = counts.get(num, 0) + 1

        thing = sorted(counts, key=counts.get, reverse=True)

        return thing[:k]